source/code/source/playfab/PlayFabApiSettings.cpp

```diff
@@ -5,39 +5,33 @@
 
 namespace PlayFab
 {
+    const std::map<std::string, std::string> PlayFabApiSettings::requestGetParams = {
+        { "sdk", PlayFabSettings::versionString }
+    };
+
     PlayFabApiSettings::PlayFabApiSettings() :
-#ifndef DISABLE_PLAYFABCLIENT_API
-        advertisingIdType(PlayFabSettings::advertisingIdType),
-        advertisingIdValue(PlayFabSettings::advertisingIdValue),
-        disableAdvertising(PlayFabSettings::disableAdvertising),
-#endif
-        verticalName(PlayFabSettings::verticalName),
         baseServiceHost(PlayFabSettings::productionEnvironmentURL),
-        titleId(PlayFabSettings::titleId)
+        titleId()
     {
+        // Don't let PlayFabSettings::staticSettings pull titleId from itself
+        if (PlayFabSettings::staticSettings != nullptr)
```

Comment:
in your latest commit, the shared_ptr is const, static, and initialized as non-null so I think this if-check is unnecessary. It should always be non-null

Reply:
This is not QUITE true, because this constructor is called when that object is still being constructed.
That's why I had to add the check to begin with.

---
In reply to: [343379329](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343379329) [](ancestors = 343379329)